tests_e2e/tests/agent_status/agent_status.py

```diff
@@ -0,0 +1,143 @@
+    is_valid = True
+
+    # Validate vm_agent_version field
+    vm_agent_version = instance_view.vm_agent.vm_agent_version
+    if vm_agent_version is None:
+        is_valid = False
+        log.info("Instance view is missing agent version, waiting to retry...")
+    elif 'Unknown' in vm_agent_version:
+        is_valid = False
+        log.info("Instance view shows agent version is unknown, waiting to retry...")
+
+    # Validate statuses field
+    statuses = instance_view.vm_agent.statuses
+    if statuses is None:
+        is_valid = False
+        log.info("Instance view is missing agent statuses, waiting to retry...")
+    elif len(statuses) < 1:
+        is_valid = False
+        log.info("Instance view is missing an agent status entry, waiting to retry...")
+    else:
+        is_valid = validate_instance_view_vmagent_status(instance_view=instance_view)
+
+    return is_valid
+
+
+def validate_instance_view(instance_view: VirtualMachineInstanceView):
+    instance_view_is_valid = True
+
+    if instance_view.vm_agent is None:
+        instance_view_is_valid = False
+        log.info("Instance view is missing vm agent, waiting to retry...")
+    else:
+        instance_view_is_valid = validate_instance_view_vmagent(instance_view=instance_view)
+
+    if instance_view.statuses is None:
+        instance_view_is_valid = False
+        log.info("Instance view is missing statuses, waiting to retry...")
+
+    if instance_view_is_valid:
+        log.info("Instance view is valid, agent version: {0}, status: {1}"
+                 .format(instance_view.vm_agent.vm_agent_version,
+                         instance_view.vm_agent.statuses[0].display_status))
+
+    return instance_view_is_valid
+
+
+class AgentStatus(AgentTest):
+    def __init__(self, context: AgentTestContext):
+        super().__init__(context)
+        self._ssh_client = SshClient(
+            ip_address=self._context.vm_ip_address,
+            username=self._context.username,
+            private_key_file=self._context.private_key_file)
+
+    def run(self):
+        log.info("")
+        log.info("*******Verifying the agent status*******")
+
+        vm = VirtualMachineClient(self._context.vm)
+
+        timeout = datetime.datetime.now() + datetime.timedelta(minutes=5)
+        instance_view_is_valid = False
+
+        # Retry validating instance view with timeout of 5 minutes
+        while datetime.datetime.now() < timeout and not instance_view_is_valid:
+            instance_view = vm.get_instance_view()
+            log.info("")
+            log.info("Validating VM Instance View...")
+            log.info("Instance view of VM is:\n%s", instance_view.serialize())
+
+            instance_view_is_valid = validate_instance_view(instance_view)
+            if not instance_view_is_valid:
+                log.info("")
+                log.info("Instance view is not valid, waiting 10s before retry...")
+                sleep(10)
```

Comment:
sleep 10 secs with timeout 5 mins is too aggressive retries. I don't think too frequent reties needed here as we decide to wait for 5 mins.